integration/spark2/src/main/scala/org/apache/carbondata/events/Events.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.events
+
+import org.apache.carbondata.core.metadata.CarbonTableIdentifier
+import org.apache.carbondata.processing.loading.model.CarbonLoadModel
+
+/**
+ * event for database operations
+ */
+trait DatabaseEvent extends Event {
+  val databaseName: String
+}
+
+/**
+ * event for table related operations
+ */
+trait TableEvent extends DatabaseEvent {
+  val carbonTableIdentifier: CarbonTableIdentifier
+  override lazy val databaseName: String = carbonTableIdentifier.getDatabaseName
+}
+
+/**
+ * event for load operations
+ */
+trait LoadEvent extends TableEvent {
```

Comment:
Move this trait to LoadEvents class

Reply:
I think it is better to keep all the possible events in Events.scala class in order to know what all different type of events are possible and keep their concrete implementation in different classes to clearly define their behavior.